generate.sh

```diff
@@ -2,7 +2,20 @@
 
 srcdir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )
 
+function ask1() { local yn; echo -n "$* [Y/n] "; read -r -n 1 yn; [[ "$yn" =~ [Yy] ]] && echo && return 0; [ -z "$yn" ] && return 0; echo; return 1; }
+
 for f in "$srcdir"/{,.}*.mustache; do
-    echo "Generating $(basename "$f" .mustache)..."
-    mustache meta.yml "$f" > "$(basename "$f" .mustache)"
+    target=$(basename "$f" .mustache)
+    case "$target" in
+        *.opam)
+            shortname=$(grep -e "shortname:" meta.yml | sed -e "s/^shortname: //")
+            target=${target/coq/coq-$shortname}
+            ;;
+        coq-action.yml)
+            ask1 "Do you want to create a .github/workflows subfolder for coq-action.yml?" &&
```

Comment:
Your `ask1` function is too obscure for me to understand, but what happens if the user answers no?

Reply:
If the user answers no, the one-liner returns "false" (namely a non-zero exit status: `return 1`) so in this case the `&& { code; }` snippet is not run, it is just skipped.
Basically we can use `ask1` in two ways, depending on whether we need an else clause:
* `ask1 "Question?" && { $THEN; }`
* `if ask1 "Question?"; then $THEN; else $ELSE; fi`

Reply:
so in the particular case of the suggested code, if the user answers `no`, all the `mustache` calls will be run with the default value of `$target`, but no `.github/workflows` subfolder is created.
(but if need be, we'll be able to add other similar `ask1` questions to conditionally run some of the `mustache` calls… maybe in another PR? or in this one)

Reply:
OK. Would it make sense to *not* generate the GitHub action file in the case the answer was no? (Creating the file at the root doesn't really make sense.) The question could be: do you wish to generate the file `.github/...`?

Reply:
Yes! that would make sense indeed. It suffices to add a `continue` instruction.
BTW I guess it would make sens to add a `continue` for the other branches as well (e.g. if there is no `extracted_shortname:` in the `meta.yml` file, we need not creating `extracted.opam`). I will push another commit.

Reply:
done.